packages/onchain-claims/src/claim-revocation.ts

```diff
@@ -0,0 +1,130 @@
+import { Event, utils, providers } from 'ethers';
+import { addressOf, EwSigner } from '@ew-did-registry/did-ethr-resolver';
+import {
+  ClaimsRevocationRegistry,
+  ClaimsRevocationRegistry__factory,
+} from '../ethers';
+
+export class ClaimRevocation {
+  private _revocationRegistry: ClaimsRevocationRegistry;
+
+  /**
+   * @param provider
+   * @param revocationRegistryOnChainAddr - Address of the on chain claim's RevocationRegistry
+   */
+  constructor(revoker: EwSigner, revocationRegistryOnChainAddr: string) {
+    this._revocationRegistry = ClaimsRevocationRegistry__factory.connect(
+      revocationRegistryOnChainAddr,
+      revoker
+    );
+  }
+
+  /**
+   * Revokes the on chain claim
+   * Returns true on success
+   *
+   * @param { string } namespace - namespace(role) to be revoked
+   * @param { string } subject - DID of user whose the claim is being revoked
+   * @param { string } revoker - DID of the revoker
+   * @returns Promise<boolean>
+   */
+  async revokeClaim(
+    namespace: string,
+    subject: string,
+    revoker: string
+  ): Promise<boolean> {
+    const namespaceHash = utils.namehash(namespace);
+    const subjectAddress = addressOf(subject);
+    const revokerAddress = addressOf(revoker);
+    try {
+      const tx = await this._revocationRegistry.revokeClaim(
+        namespaceHash,
+        subjectAddress,
+        revokerAddress
+      );
+      const receipt = await tx.wait();
+      const event = receipt.events?.find((e: Event) => e.event === 'Revoked');
+      if (!event) return false;
+    } catch (error) {
+      if (error instanceof Error) {
```

Comment:
What was the reason to add such check?

Reply:
If you are asking about `if (error instanceof Error) `, then it was added when we upgraded `typescript`. For some reason the error regarding the use of `any` was not getting eliminated and it is there for all such call in `ew-did-registry` if I remember correctly.

Reply:
Most of the eslint warnings are for `no-unresolved`. I would put some time and work on it later.

Reply:
Okay, why we catch error and then re-throw it?